docs/GroupDefs.hpp

```diff
@@ -159,155 +164,485 @@
  * Charm++
 
 SpECTRE builds a layer on top of Charm++ that performs various safety checks and
-initialization for the user that can otherwise lead to difficult to debug
+initialization for the user that can otherwise lead to difficult-to-debug
 undefined behavior. The central concept is what is called a %Parallel
 Component. A %Parallel Component is a struct with several type aliases that
 is used by SpECTRE to set up the Charm++ chares and allowed communication
-patterns. It might be most natural to think of %Parallel Components
-as input arguments to a metaprogram that writes the parallelization
-infrastructure that you requested for the executable. There is no restriction
-on the number of %Parallel Components, though practically it is best to have
-around 10 at most.
-
-Each %Parallel Component must have the following type aliases:
-1. `using chare_type` is set to one of `Parallel::Algorithms::Singleton`,
-   `Parallel::Algorithms::Array`, `Parallel::Algorithms::Group`, or
-   `Parallel::Algorithms::Nodegroup`. What these mean is explained below.
+patterns. %Parallel Components are input arguments to the compiler, which then
+writes the parallelization infrastructure that you requested for the executable.
+There is no restriction on the number of %Parallel Components, though
+practically it is best to have around 10 at most.
+
+### The Metavariables Class
+
+SpECTRE takes a different approach to input options passed to an executable than
+is common. SpECTRE not only reads an input file at runtime but also has many
+choices made at compile time. The compile time options are specified by what is
+referred to as the metavariables. What exactly the metavariables specifies
+depends somewhat on the executable, but all metavariables structs must specify
+the following:
+
+- `help`: a `static constexpr OptionString` that will be printed as part of the
+  help message. It should describe the executable and basic usage of it, as well
+  as any non-standard options that must be specified in the metavariables and
+  their current values. An example of a help string for one of the testing
+  executables is:
+  \snippet Test_AlgorithmCore.cpp help_string_example
+- `component_list`: a `tmpl::list` of the parallel components (described below)
+  that are to be created. Most evolution executables will have the
+  `DgElementArray` parallel component listed. An example of a `component_list`
+  for one of the test executables is:
+  \snippet Test_AlgorithmCore.cpp component_list_example
```

Comment:
Perhaps show a snippet of `EvolveScalarWave` instead, that is closer to common use cases?

Reply:
We don't allow snippets from inside the production code currently. The reason being that production code may not be compiled, whereas the tests will always be compiled and tested.

I'm happy to add more of an explanation, what do you think should be covered?

Reply:
Yeah let’s do that, let’s pick a particular example (`EvolveScalarWave`) and document it heavily.

Instead of the snippet you could briefly describe a common use case, e.g. have one parallel component perform an actions sequence on an array of elements, and another to write data to disk. This gives a more intuitive picture of what these structures are intended for.